"""
@generated by mypy-protobuf.  Do not edit manually!
isort:skip_file
"""
import builtins
import google.protobuf.descriptor
import google.protobuf.internal.enum_type_wrapper
import google.protobuf.message
import typing
import typing_extensions

DESCRIPTOR: google.protobuf.descriptor.FileDescriptor = ...

class ConversionAdjustmentUploadErrorEnum(google.protobuf.message.Message):
    """Proto file describing conversion adjustment upload errors.

    Container for enum describing possible conversion adjustment upload errors.
    """
    DESCRIPTOR: google.protobuf.descriptor.Descriptor = ...
    class _ConversionAdjustmentUploadError:
        ValueType = typing.NewType('ValueType', builtins.int)
        V: typing_extensions.TypeAlias = ValueType
    class _ConversionAdjustmentUploadErrorEnumTypeWrapper(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[_ConversionAdjustmentUploadError.ValueType], builtins.type):
        DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor = ...
        UNSPECIFIED: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 0
        """Not specified."""

        UNKNOWN: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 1
        """The received error code is not known in this version."""

        TOO_RECENT_CONVERSION_ACTION: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 2
        """The specified conversion action was created too recently.
        Please try the upload again after 4-6 hours have passed since the
        conversion action was created.
        """

        INVALID_CONVERSION_ACTION: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 3
        """No conversion action of a supported ConversionActionType that matches the
        provided information can be found for the customer.
        """

        CONVERSION_ALREADY_RETRACTED: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 4
        """A retraction was already reported for this conversion."""

        CONVERSION_NOT_FOUND: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 5
        """A conversion for the supplied combination of conversion
        action and conversion identifier could not be found.
        """

        CONVERSION_EXPIRED: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 6
        """The specified conversion has already expired. Conversions expire after 55
        days, after which adjustments cannot be reported against them.
        """

        ADJUSTMENT_PRECEDES_CONVERSION: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 7
        """The supplied adjustment date time precedes that of the original
        conversion.
        """

        MORE_RECENT_RESTATEMENT_FOUND: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 8
        """A restatement with a more recent adjustment date time was already
        reported for this conversion.
        """

        TOO_RECENT_CONVERSION: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 9
        """The conversion was created too recently."""

        CANNOT_RESTATE_CONVERSION_ACTION_THAT_ALWAYS_USES_DEFAULT_CONVERSION_VALUE: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 10
        """Restatements cannot be reported for a conversion action that always uses
        the default value.
        """

        TOO_MANY_ADJUSTMENTS_IN_REQUEST: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 11
        """The request contained more than 2000 adjustments."""

        TOO_MANY_ADJUSTMENTS: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 12
        """The conversion has been adjusted too many times."""

    class ConversionAdjustmentUploadError(_ConversionAdjustmentUploadError, metaclass=_ConversionAdjustmentUploadErrorEnumTypeWrapper):
        """Enum describing possible conversion adjustment upload errors."""
        pass

    UNSPECIFIED: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 0
    """Not specified."""

    UNKNOWN: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 1
    """The received error code is not known in this version."""

    TOO_RECENT_CONVERSION_ACTION: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 2
    """The specified conversion action was created too recently.
    Please try the upload again after 4-6 hours have passed since the
    conversion action was created.
    """

    INVALID_CONVERSION_ACTION: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 3
    """No conversion action of a supported ConversionActionType that matches the
    provided information can be found for the customer.
    """

    CONVERSION_ALREADY_RETRACTED: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 4
    """A retraction was already reported for this conversion."""

    CONVERSION_NOT_FOUND: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 5
    """A conversion for the supplied combination of conversion
    action and conversion identifier could not be found.
    """

    CONVERSION_EXPIRED: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 6
    """The specified conversion has already expired. Conversions expire after 55
    days, after which adjustments cannot be reported against them.
    """

    ADJUSTMENT_PRECEDES_CONVERSION: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 7
    """The supplied adjustment date time precedes that of the original
    conversion.
    """

    MORE_RECENT_RESTATEMENT_FOUND: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 8
    """A restatement with a more recent adjustment date time was already
    reported for this conversion.
    """

    TOO_RECENT_CONVERSION: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 9
    """The conversion was created too recently."""

    CANNOT_RESTATE_CONVERSION_ACTION_THAT_ALWAYS_USES_DEFAULT_CONVERSION_VALUE: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 10
    """Restatements cannot be reported for a conversion action that always uses
    the default value.
    """

    TOO_MANY_ADJUSTMENTS_IN_REQUEST: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 11
    """The request contained more than 2000 adjustments."""

    TOO_MANY_ADJUSTMENTS: ConversionAdjustmentUploadErrorEnum.ConversionAdjustmentUploadError.ValueType = ...  # 12
    """The conversion has been adjusted too many times."""


    def __init__(self,
        ) -> None: ...
global___ConversionAdjustmentUploadErrorEnum = ConversionAdjustmentUploadErrorEnum
